br/pkg/checksum/executor.go

```diff
@@ -330,13 +332,30 @@ func (exec *Executor) Execute(
 	updateFn func(),
 ) (*tipb.ChecksumResponse, error) {
 	checksumResp := &tipb.ChecksumResponse{}
+	checksumBackoffer := utils.NewFixedBackoffer(utils.ChecksumRetryTime, utils.ChecksumWaitInterval, utils.ChecksumMaxWaitInterval)
 	for _, req := range exec.reqs {
 		// Pointer to SessionVars.Killed
 		// Killed is a flag to indicate that this query is killed.
 		//
 		// It is useful in TiDB, however, it's a place holder in BR.
 		killed := uint32(0)
-		resp, err := sendChecksumRequest(ctx, client, req, kv.NewVariables(&killed))
+		var (
+			resp *tipb.ChecksumResponse
+			err  error
+		)
+		err = utils.WithRetry(ctx, func() error {
+			resp, err = sendChecksumRequest(ctx, client, req, kv.NewVariables(&killed))
+			failpoint.Inject("checksumRetryErr", func(val failpoint.Value) {
+				// first time reach here. return error
+				if val.(bool) && checksumBackoffer.Attempt() == utils.ChecksumRetryTime {
+					err = errors.New("inject checksum error")
+				}
```

Comment:
...so we no longer need to check the attempt time here. Also, the boolean argument may be removed too.
```suggestion
					err = errors.New("inject checksum error")
```